lightningd/datastore.c

```diff
@@ -77,6 +77,10 @@ static struct command_result *param_list_or_string(struct command *cmd,
 							     "should be string");
 			(*str)[i] = json_strdup(*str, buffer, t);
 		}
+	} else if (tok-> type == JSMN_ARRAY && tok->size == 0){
+		return command_fail_badparam(cmd, name,
+								buffer, tok,
+								"should not be empty");
```

Comment:
```suggestion
	} else if (tok-> type == JSMN_ARRAY && tok->size == 0){
		return command_fail_badparam(cmd, name,
								buffer, tok,
								"should not be empty");
```

as a minor comment I would like to get this check as first if, imho it is more clear to read.

Reply:
So you prefer the checks in this order?
 - is empty array
 - is non-empty array
 - is string or primitive
 - else

Reply:
yeah, sounds easier to read?